target/linux/realtek/files-6.12/drivers/net/dsa/rtl83xx/rtl930x.c

```diff
@@ -2272,11 +2274,20 @@ static void rtl930x_led_init(struct rtl838x_switch_priv *priv)
 		sprintf(set_name, "led_set%d", set);
 		leds_in_this_set = of_property_count_u32_elems(node, set_name);
 
-		if (leds_in_this_set == 0 || leds_in_this_set > sizeof(set_config)) {
-			pr_err("%s led_set configuration invalid skipping over this set\n", __func__);
+		if (leds_in_this_set == -EINVAL) {
+			// if this led_setX property is not present, skip it silently
 			continue;
 		}
 
+		if (leds_in_this_set <= 0 || leds_in_this_set > sizeof(set_config)) {
+			pr_err("%s: %s configuration invalid skipping over this set, leds_in_this_set=%d (should be >0 and <=%d\n",
+				__func__, set_name, leds_in_this_set, sizeof(set_config));
```

Comment:
Please use dev_err/dev_info instead of pr_XXX("%s ...", __func__);

Reply:
I'll update all the references in this function.